File: Programmers/codingTest_Kit/DFS_BFS/43162.cpp
#include <string>
#include <vector>
#define MAX 201
using namespace std;

bool Visit[MAX];
vector<int> vec[MAX];

void DFS(int X){
    for(int i = 0; i < vec[X].size(); i++){
        int node = vec[X][i];
        if(!Visit[node]){
            Visit[node] = true;
            DFS(node);
        }
    }
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    for(int i = 0; i < computers.size(); i++){
        for(int j = 0; j < n; j++){
            if(i != j){
                if(computers[i][j] == 1){
                    vec[i].push_back(j);
                }
            }
        }
    }

    for(int i = 0; i < n ; i++){
        if(!Visit[i]) {
            DFS(i);
            answer++;
        }
    }
    return answer;
}

File: Programmers/codingTest_Kit/DFS_BFS/43162.java
package Programmers.codingTest_Kit.DFS_BFS;

class Solution {

    public boolean[] VISIT = new boolean[201];

    public void DFS(int n, int[][] computers, int idx){
        VISIT[idx] = true;
        for(int i = 0; i < n; i++){
            if(i != idx && computers[idx][i] == 1){
                if(VISIT[i] == false){
                    DFS(n, computers, i);
                }
            }
        }

    }

    public int solution(int n, int[][] computers) {
        int answer = 0;
        for(int i = 0; i < n; i++){
            if(VISIT[i] == false){
                DFS(n, computers, i);
                answer++;
            }
        }
        return answer;
    }
}